test/Chainweb/Test/Pact/PactInProcApi.hs

```diff
@@ -1200,6 +1204,32 @@ txResult msg i o = do
 cbResult :: PayloadWithOutputs -> IO (CommandResult Hash)
 cbResult o = decodeStrictOrThrow @_ @(CommandResult Hash) (_coinbaseOutput $ _payloadWithOutputsCoinbase o)
 
+blockGasLimitTest :: IO (IORef MemPoolAccess) -> IO (PactQueue, TestBlockDb) -> TestTree
+blockGasLimitTest _ reqIO = testCase "blockGasLimitTest" $ do
+  (q,_) <- reqIO
+
+  let
+    useGas g = do
+      bigTx <- buildCwCmd $ set cbGasLimit g $ set cbSigners [mkSigner' sender00 []] $ mkCmd "cmd" $ mkExec' "TESTING"
+      let cr = CommandResult (RequestKey (Hash "0")) Nothing (PactResult $ Left $ PactError EvalError (Pact.Types.Info.Info $ Nothing) [] mempty) (fromIntegral g) Nothing Nothing Nothing []
+      let block = Transactions (V.singleton (bigTx, cr)) (CommandResult (RequestKey (Hash "h")) Nothing (PactResult $ Right $ PLiteral (LString "output")) 0 Nothing Nothing Nothing [])
+      let payload = toPayloadWithOutputs noMiner block
+      let bh = newBlockHeader
+                mempty
+                (_payloadWithOutputsPayloadHash payload)
+                (Nonce 0)
+                (BlockCreationTime $ Time $ TimeSpan 0)
+                (ParentHeader $ genesisBlockHeader testVersion cid)
+      validateBlock bh (payloadWithOutputsToPayloadData payload) q >>= takeMVar
+  -- we consume more than the maximum block gas limit and provoke an error.
+  useGas 2_000_001 >>= \case
+    Left (PactInternalError (decode . BL.fromStrict . T.encodeUtf8 -> Just (BlockGasLimitExceeded _))) -> return ()
+    r -> error $ "not a BlockGasLimitExceeded error: " <> sshow r
+  -- we consume exactly the maximum block gas limit and expect no such error.
+  useGas 2_000_000 >>= \case
+    Left (PactInternalError (decode . BL.fromStrict . T.encodeUtf8 -> Just (BlockGasLimitExceeded _))) -> error "consumed exactly block gas limit but errored"
+    _ -> return ()
+
```

Comment:
Maybe add a few more cases, just to make the test a bit more robust against obscure arithmetic errors? For instance:

* 1_000_000
* 0
* 1
* 3_000_000